dustcloud/server.py

```diff
@@ -31,528 +28,527 @@
-		
-	def on_read_cloud(self):
-		data = self.sock.recv(32)  # wait to get the first 32 bytes (header+md5)
-		print("{} via tcp wrote:".format(self.client_address[0]))
-		# print("C> RAW: %s" % binascii.hexlify(data))
-		if len(data) < 32: # should never receive anything smaller than 32 byte
-			print("len < 32")
-			self.sock.close()
-			return
-		else:
-			if data[0:2] == bytes.fromhex("2131"):  # Check for magic bytes
-				print("magic ok")
-				packetlenght = int.from_bytes(data[2:4],byteorder='big'); # get packet lenght from header
-				# print("packetlength %s, got already %s " % (packetlenght,len(data)))
-				if len(data) != packetlenght: # packet longer than 32 byte
-					data += self.sock.recv((packetlenght-32)) # get the rest of the packet
-					# print("packetlength %s, got now %s " % (packetlenght,len(data)))
-					# print("= RAW: %s" % binascii.hexlify(data))
-					self.Cloudi.do_log_raw(self.ddid,binascii.hexlify(data), "from_cloud")
-				process_result = self.Cloudi.process_cloud_data(self,data)
-				#if process_result == 1:
-					#self.request.close()
-					#return
-		
-	def on_read(self):
-		data = self.request.recv(32)  # wait to get the first 32 bytes (header+md5)
-		print("{} via tcp wrote:".format(self.client_address[0]))
-		#print("> RAW: %s" % binascii.hexlify(data))
-		if len(data) < 32: # should never receive anything smaller than 32 byte
-			print("len < 32")
-			self.request.close()
-			return
-		else:
-			if data[0:2] == bytes.fromhex("2131"):  # Check for magic bytes
-				print("magic ok")
-				packetlenght = int.from_bytes(data[2:4],byteorder='big'); # get packet lenght from header
-				# print("packetlength %s, got already %s " % (packetlenght,len(data)))
-				if len(data) != packetlenght: # packet longer than 32 byte
-					data += self.request.recv((packetlenght-32)) # get the rest of the packet
-					# print("packetlength %s, got now %s " % (packetlenght,len(data)))
-					# print("= RAW: %s" % binascii.hexlify(data))
-				process_result = self.Cloudi.process_data(self,data)
-				if process_result == 1:
-					self.request.close()
-					return
-			else:
-				self.httpmode=1
-				print("Magic bytes were missing, echo http redirection")
-				self.http(data)
-				http = "HTTP/1.1 302 Found\n"
-				http = http + "Location: https://xxx/\n"
-				http = http + "Content-Length: 212\n"
-				http = http + "Content-Type: text/html; charset=iso-8859-1\n"
-				http = http + "\n"
-				http = http + "<!DOCTYPE HTML PUBLIC \"-//IETF//DTD HTML 2.0//EN\">\n"
-				http = http + "<html><head>\n"
-				http = http + "<title>302 Found</title>\n"
-				http = http + "</head><body>\n"
-				http = http + "<h1>Found</h1>\n"
-				http = http + "<p>The document has moved <a href=\"https://xxx/\">here</a>.</p>\n"
-				http = http + "</body></html>"
-				# self.request.send(http.encode('utf-8'))
-				self.request.close()
-				return
-				
-	def http(self,firstdata):
-		server_address = ('localhost', 5080)
-		self.sock.connect(server_address)

```

Comment:
Not sure what this http stuff is supposed to do but it tries to connect to localhost:5080 which always fails because there is nothing running there (at least for me). As this will raise an exception and basically abort the connection when receiving unknown messages I removed it completely.

Reply:
Hi,
the idea here is to forward traffic to a real http server (as we are listening on port 80). This is a feature for specific devices, that get specific data over http...so that you do not need 2 different ip addresses (if you host on a public server).

Reply:
Will add this back in behind a command line flag